### ISSUE TITLE
impliment .return

### ISSUE THREAD

**Issue by calvinmetcalf** [calvinmetcalf/async-iter-stream#5]
how should we handle this, just with destroy?

**Comment by mcollina** [comment]
Yes, I think `.destroy()` would be ok. However, we need a way to return without destroying.

**Comment by calvinmetcalf** [comment]
> However, we need a way to return without destroying.

do you mean we need a way to what, destroy the iterator without destroying the stream?
